nesta/core/routines/datasets/nih/nih_dedupe_task.py

```diff
@@ -1,131 +1,75 @@
-# TODO: update anything here with latest method (e.g. mysqltarget)
-# TODO: set default batchable and runtime params where possible
-# TODO: update orm, where required, incl lots of indexes
-# TODO: update batchable as required
-# TODO: write decent tests to check good dq
 '''
 Deduplication of near duplicates
 ================================
 
-Remove nears duplicates of projects from the data. Numeric
-fields (such as funding) are aggregated together.
+Discover near-duplicates of projects from similarity 
+of document vectors, and fill out the duplicates link table.
 '''
 
 import logging
 import luigi
-import datetime
+from datetime import datetime as dt
 
-from nesta.packages.misc_utils.batches import split_batches, put_s3_batch
-from nesta.core.luigihacks.misctools import find_filepath_from_pathstub as f3p
-from nesta.core.luigihacks.mysqldb import MySqlTarget
-from nesta.core.luigihacks import autobatch
-from nesta.core.orms.orm_utils import get_es_ids
-from nesta.core.orms.orm_utils import setup_es
-from nesta.core.orms.orm_utils import get_config
+from nesta.core.luigihacks.misctools import f3p
+from nesta.core.luigihacks.mysqldb import make_mysql_target
+from nesta.core.luigihacks.parameter import SqlAlchemyParameter
+from nesta.core.orms.nih_orm import AbstractVector
+from nesta.core.orms.nih_orm import PhrVector
+from nesta.core.orms.nih_orm import TextDuplicate
+from nesta.core.orms.nih_orm import Base
+from nesta.core.orms.orm_utils import insert_data
 
-from nesta.core.routines.nih.nih_data.nih_abstracts_mesh_task import AbstractsMeshTask
+from nesta.packages.vectors.similarity import generate_duplicate_links
+import faiss
 
-class DedupeTask(autobatch.AutoBatchTask):
-    '''
+
+class VectorDedupeTask(luigi.Task):
+    '''Generic task for generating a link table duplicate text fields
```

Comment:
Exactly, there are elements of it which are generic but this dataset is already an exceptional case because there are two text fields in two separate tables which could indicate being a duplicate. Perhaps with more use-cases it will be easier to factor out commonalities, but in the meanwhile the `generate_duplicate_links` is pretty generic